helm/cluster-cloud-director/templates/_helpers.tpl

```diff
@@ -219,3 +220,65 @@ taints:
 {{- $outerScope := . }}
 {{- include "mtRevision" (merge (dict "currentClass" .Values.controlPlane) $outerScope.Values) }}
 {{- end -}}
+
+{{/*
+Join elements of a string array by quoting elements and using comma as seperator.
+*/}}
+{{- define "commaJoinedQuotedList" -}}
+{{- $list := list }}
+{{- range . }}
+{{- $list = append $list (printf "\"%s\"" .) }}
+{{- end }}
+{{- join ", " $list }}
+{{- end }}
+
+{{/*
+Generate a stanza for KubeAdmConfig and KubeAdmControlPlane in order to 
+mount containerd configuration for registry configuration in nodes.
+*/}}
+{{- define "registryFiles" -}}
+{{- if .Values.registry.configure -}}
+- path: /etc/containerd/conf.d/registry-config.toml
+  permissions: "0600"
+  contentFrom:
+    secret:
+      name: {{ include "registrySecretName" $ }}
+      key: registry-config.toml
+{{- end -}}
+{{- end -}}
```

Comment:
Should we not terminate `{{- if .Values.registry.configure -}}` at the end `(line 284`) ?
And move `{{- if .Values.registry.configure -}}` above `{{- define "registryFiles" -}}`

Reply:
I didn't get your comment. Can you elaborate a little more?

Reply:
I added suggestions of what I mean.